src/model/warning.ts

```diff
@@ -34,3 +38,35 @@ export class WarningCarryingException extends Error {
     this.warning = warning;
   }
 }
+
+const emptySet = new Set();
```

Comment:
bah, sharing a single set like this makes my skin crawl. Wish there was a way to freeze a set so that we could guarantee it was empty. oh well.